hs628_tools/regression_check.py

```diff
@@ -0,0 +1,189 @@
+# coding: utf-8
+
+import numpy as np
+import pandas as pd
+import sklearn as sk
+from sklearn.linear_model import LinearRegression
+import statsmodels.api as sm
+import matplotlib.pyplot as plt
+import seaborn as sns
+import os
+import sys
+# %config IPCompleter.greedy=True
+# %matplotlib inline
+
+#   ****** Data Visualization to check Linear regression assumptions and calculating new regression metric ******
+
+class RegCheck():
+    """
+        This Class is created for checking Linear Regression assumptions and Evaluating the Model
+        The following functions are defined in this class :
+
+         func : check_linearity_residual_homoscedasticity_independence_plt checks to see if the
+                data can fit to a linear model/non-linear model. Also check the homoscedasticity
+                (check for the spread of data) of the residuals/error terms and also check for their
+                independence. We diagnose these through a Residuals vs Predicted values plot.If there
+                is no clear pattern and data symmetrically distributed it can be said to satisfy the
+                assumptions.
+
+         func : check_residuals_normality_plt checks to see the normality of the error terms through
+                a density plot of the residuals. If the graph looks like gaussian normal curve it can
+                be said that the residuals satisfy the assumption of normality.
+
+         func : check_predictors_independence_plt checks for the independence of each predictor by
+                plotting a heatmap of the correlations of each predictor with one and another. The
+                heatmap would clearly indicate if there are multicolinear predictors.
+
+         func : regression_metric builds a metric for evaluating the model. It is a harmonic mean of
+                the R-Squared value and the F-Statistic taking into consideration the probability of
+                significance of the F-statistic being significant than having an only intercept model.
+    """
+
+    def __init__(self):
+        # Instantiating the parameters used in the methods below :PredictorColumnnames,R2,
+        # F-stat,F-statpval,ResidualSumofSquares,RegressionMetric
+        self.colnames = None
+        self.rsquared = 0
+        self.fstatistic = 0
+        self.fstat_pval = 0
+        self.resid_sumsq = 0
+        self.regmetric = 0
+
+    def check_linearity_residual_homoscedasticity_independence_plt(self, X, y, model):
+        """
+        Checks to see if the data can fit to a linear model/non-linear model,checks the homoscedasticity
+         of the residuals/error terms and also checks for their independence.
+            -----------
+            Param:
+                X: 2-D numpy array of the predictors
+                y: 1-D numpy array of the response variables
+                linear : Linear regression model fitted to X(predictors) and y(response)
```

Comment:
It looks like this parameter name has been updated to instead be 'model' (which I do agree is a better name).  Would be best to also note in the docstring that the code requires this to have a ```predict``` method (like an sklearn or statsmodels linear regression model would).

Reply:
Will be changing the parameter name in docstrings, did forget to change it. But I'm not entirely sure what you mean by 'note in the docstring that the code requires this to have a predict method '. Could you explain more about this as we just fit the data to the model, we do not actually use predict on the model anywhere here.

Reply:
> But I'm not entirely sure what you mean by 'note in the docstring that the code requires this to have a predict method '. Could you explain more about this as we just fit the data to the model, we do not actually use predict on the model anywhere here.


[This is the use of 'predict'](https://github.com/EHWUSF/HS68_2018_Project_1/pull/31#discussion_r208818489) I was referring to

Reply:
Oh I see my bad, I thought the comment was meant for a line in docstrings. But now I see you meant the fact that we are predicting values later in the function needs to be mentioned in. Okay I understand it, thank you.